XMonad/Prompt.hs

```diff
@@ -410,9 +410,10 @@ highlightedItem st' completions = case complWinDim st' of
       (_,_,_,_,xx,yy) = winDim
       complMatrix = splitInSubListsAt (length yy) (take (length xx * length yy) completions)
       (col_index,row_index) = complIndex st'
-    in case completions of
-      [] -> Nothing
-      _  -> complMatrix !? col_index >>= (!? row_index)
+    in case length completions of
+      0 -> Nothing
+      1 -> Just $ complMatrix !! col_index !! row_index
```

Comment:
This seems really weird. Whenever `complMatrix !? col_index >>= (!? row_index)` returns `Just`, this behaves the same, and when it would return `Nothing`, this one is a bottom. Or am I missing something?

Reply:
Fixed in 344ace492b5e03769ccea334be78661b411c465d

Reply:
:-)

I'll just trust on this one then, I know very little about the Prompt internals and nothing really weird stands out any more.